File: app/src/main/java/com/adventure/solo/SoloAdventureApp.java
package com.adventure.solo;

import android.app.Application;
import dagger.hilt.android.HiltAndroidApp;
import org.osmdroid.config.Configuration;

@HiltAndroidApp
public class SoloAdventureApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        // Initialize OpenStreetMap configuration
        Configuration.getInstance().setUserAgentValue(getPackageName());
    }
} 

File: app/src/main/java/com/adventure/solo/SoloAdventureApp.kt
package com.adventure.solo

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
import org.osmdroid.config.Configuration

@HiltAndroidApp
class SoloAdventureApp : Application() {
    override fun onCreate() {
        super.onCreate()
        // Initialize OpenStreetMap configuration
        Configuration.getInstance().userAgentValue = packageName
    }
} 